Server/mods/deathmatch/logic/luadefs/CLuaResourceDefs.cpp

```diff
@@ -1468,3 +1470,23 @@ bool CLuaResourceDefs::isResourceProtected(CResource* const resource)
 {
     return resource->IsProtected();
 }
+
+std::vector<std::string> CLuaResourceDefs::GetLoadedFiles(lua_State* luaVM, std::optional<CResource*> resource) noexcept
+{
+    if (!resource)
+        resource = &lua_getownerresource(luaVM);
+
+    const auto resourceFiles = (*resource)->GetResourceFiles();
+
+    std::vector<std::string> files;
+    files.reserve(resourceFiles.size());
+    for (const auto& file : resourceFiles)
+    {
+        if (file->GetType() != CResourceFile::eResourceType::RESOURCE_FILE_TYPE_CLIENT_FILE)
```

Comment:
Are you sure that a server side function should deal with RESOURCE_FILE_TYPE_CLIENT_FILE files?

Reply:
Maybe rename the function to `getLoadedClientFiles` for clarity?

Reply:
> Maybe rename the function to `getLoadedClientFiles` for clarity?

Doesnt matter. `RESOURCE_FILE_TYPE_CLIENT_FILE` just means a valid `<file>` field. It will return every file thats not a map and not a script.

Reply:
And please, do not mark a conversation resolved if the actual conversastion is in process.

Reply:
> > `<file>` field means exactly a client file
> 
> thats the only file than can be. There is no division between server files and client files because server files is just not a concept.

Exactly for this reason the function should be called `getLoadedClientFiles`(or maybe `getResourceClientFileNames`). Simply because this function cannot deal with server-side files.

Reply:
Renaming it to `getLoadedClientFiles` would indicate that there is a `getLoadedServerFiles` function. That doesnt make sense because server-side files dont exist.

Reply:
> Renaming it to `getLoadedClientFiles` would indicate that there is a `getLoadedServerFiles` function. That doesnt make sense because server-side files dont exist.

The existence of `getLoadedClientFiles` function doesn't lead to the existence of function `getLoadedServerFiles`.

Reply:
Anyway, this function must be renamed(maybe you will come up with a better name). Because the current name is confusing.

Reply:
> > Anyway, this function must be renamed(maybe you will come up with a better name). Because the current name is confusing.
> 
> I disagree. Instead of renaming it, add a string argument that lets you decide if it should return files, map files, script files, or all. Like "all"/"files"/"scripts"/"maps" according to the xml tags in meta.xml https://wiki.multitheftauto.com/wiki/Meta.xml

This is what I'm also come up with. But you got ahead of me. Yes, this will work.

Reply:
This argument can be optional and default to "all" where it returns script, map and regular file paths all together in a table.

Reply:
`getResourceFiles` is explicit. The name of this function should start with 'getResource' because this function related to resource class. And the function can be shared with this name

Reply:
I agree. That is a more suiting name for this function.